bin/metrics-cassandra-graphite.rb

```diff
@@ -347,7 +347,8 @@ def get_metric(string) # rubocop:disable NestedMethodDefinition
         metric.gsub!(/[_]{2,}/, '_')       # convert sequence of multiple _'s to single _
         metric.downcase!
         # sanitize metric values for graphite. Numbers only, please.
-        value = value.chomp(' ms.').gsub(/([0-9.]+)$/, '\1')
+        # some versions of nodetool omit the '.' following the 'ms' unit.
+        value = value.chomp(' ms.').chomp(' ms').gsub(/([0-9.]+)$/, '\1')
```

Comment:
What do the strings look like? Doing 2 `chomp`s in a row makes followed by a `gsub` me think there is a more efficient way to accomplish this.

Reply:
The issue is that the old string looked like this:
`Read Latency: 1.4066805555555557 ms.`
whereas the new string would be
`Read Latency: 1.4066805555555557 ms`

It is literally the same, minus the period at the end.

There may be a more efficient way to do this, however to my knowledge, Ruby's 'chomp' only works with a single field separator and does not allow for something like "match on zero or one instances of `.`

I tested my code against both formats and it works.  Obviously if you swap the chomp order you will be left with a hanging `.` if nodetool is returning values with the period.

Reply:
I see, also I am not quite sure I understand the meaning/need of that gsub. From my testing it seems unneeded:
```
# create some variables for testing
irb(main):001:0> string1 = 'Read Latency: 1.4066805555555557 ms.'
=> "Read Latency: 1.4066805555555557 ms."
irb(main):002:0> string2 = 'Read Latency: 1.4066805555555557 ms'
=> "Read Latency: 1.4066805555555557 ms"

# testing both strings
irb(main):004:0> string2.split(/(\.)?(ms)(\.)?/).first.gsub(/([0-9.]+)$/, '\1')
=> "Read Latency: 1.4066805555555557 "

irb(main):006:0> string2.split(/(\.)?(ms)(\.)?/).first
=> "Read Latency: 1.4066805555555557 "
```

I am also noticing a trailing space on the value returned. Can you test if this works? It removes the trailing string and handles both cases with what I _think_ is the right output:
```
irb(main):005:0> string2.split(/(\.)?(ms)(\.)?/).first.gsub(/\s$/, '')
=> "Read Latency: 1.4066805555555557"

irb(main):008:0> string1.split(/(\.)?(ms)(\.)?/).first.gsub(/\s$/, '')
=> "Read Latency: 1.4066805555555557"
```

I did a quick bendmark to see what the difference was and it appears that doing the 2 `.chomp`s are indeed faster than using a regex and split. While the performance is negligible I was curious to see what removing the final gsub did performance wise when it looked like it was not needed:
```ruby
# with the gsub at the end
irb(main):018:0> puts Benchmark.measure {string2.chomp(' ms.').chomp(' ms').gsub(/([0-9.]+)$/, '\1')}
  0.000000   0.000000   0.000000 (  0.000018)

irb(main):019:0> puts Benchmark.measure {string2.chomp(' ms.').chomp(' ms').gsub(/([0-9.]+)$/, '\1')}
  0.000000   0.000000   0.000000 (  0.000013)

irb(main):020:0> puts Benchmark.measure {string2.chomp(' ms.').chomp(' ms').gsub(/([0-9.]+)$/, '\1')}
  0.000000   0.000000   0.000000 (  0.000020)

# without the gsub at the end
irb(main):021:0> puts Benchmark.measure {string2.chomp(' ms.').chomp(' ms')}
  0.000000   0.000000   0.000000 (  0.000007)

irb(main):022:0> puts Benchmark.measure {string1.chomp(' ms.').chomp(' ms')}
  0.000000   0.000000   0.000000 (  0.000005)

irb(main):023:0> puts Benchmark.measure {string1.chomp(' ms.').chomp(' ms')}
  0.000000   0.000000   0.000000 (  0.000005)

# replacing the unneeded gsub with one that removes the trailing space put us essentially back where we were
irb(main):024:0> puts Benchmark.measure {string1.chomp(' ms.').chomp(' ms').gsub(/\s$/, '')}
  0.000000   0.000000   0.000000 (  0.000020)

irb(main):025:0> puts Benchmark.measure {string1.chomp(' ms.').chomp(' ms').gsub(/\s$/, '')}
  0.000000   0.000000   0.000000 (  0.000016)

irb(main):026:0> puts Benchmark.measure {string1.chomp(' ms.').chomp(' ms').gsub(/\s$/, '')}
  0.000000   0.000000   0.000000 (  0.000016)
```

If the gsub is not needed I'd say either we should remove it (to simply code rather than performance) and if we want to make it a little bit cleaner we should replace it with just removing the trailing space from the output. I'd accept it at this point either way but I always love taking the opportunity to reduce code complexity.

Reply:
I am not getting the spurious space at the end...

```irb(main):012:0> string1 = 'Read Latency: 1.4066805555555557 ms'
=> "Read Latency: 1.4066805555555557 ms"
irb(main):013:0> string2 = 'Read Latency: 1.4066805555555557 ms.'
=> "Read Latency: 1.4066805555555557 ms."
irb(main):014:0> string1.chomp(' ms.').chomp(' ms')
=> "Read Latency: 1.4066805555555557"
irb(main):015:0> string2.chomp(' ms.').chomp(' ms')
=> "Read Latency: 1.4066805555555557"
```

Reply:
The gsub confused me as well, and I cannot see a purpose for it.

Reply:
> I am not getting the spurious space at the end...
> 
> ```
> => "Read Latency: 1.4066805555555557 ms"
> irb(main):013:0> string2 = 'Read Latency: 1.4066805555555557 ms.'
> => "Read Latency: 1.4066805555555557 ms."
> irb(main):014:0> string1.chomp(' ms.').chomp(' ms')
> => "Read Latency: 1.4066805555555557"
> irb(main):015:0> string2.chomp(' ms.').chomp(' ms')
> => "Read Latency: 1.4066805555555557"
> ```

I will chalk that up to me mapybe cop/pasting that wrong.

> The gsub confused me as well, and I cannot see a purpose for it.

Ya I'd say lets remove it, it its reasoning escapes me.